src/lib/authentification.ts

```diff
@@ -230,33 +197,58 @@ export function fetchActors(agent?: HttpAgent, isPlug = false): Promise<Actors>
 				});
 			}
 
-			const icpLedger: icpLedgerInterface = Actor.createActor(idlFactoryIcp, {
+			const icpLedger = new CanisterActor<icpLedgerInterface>(
 				agent,
-				canisterId: CANISTER_ID_ICP_LEDGER
-			});
-
-			const nicpLedger: icrcLedgerInterface = Actor.createActor(idlFactoryIcrc, {
+				idlFactoryIcp,
+				CANISTER_ID_ICP_LEDGER
+			);
+			const nicpLedger = new CanisterActor<icrcLedgerInterface>(
 				agent,
-				canisterId: CANISTER_ID_NICP_LEDGER
-			});
-			const wtnLedger: icrcLedgerInterface = Actor.createActor(idlFactoryIcrc, {
+				idlFactoryIcrc,
+				CANISTER_ID_NICP_LEDGER
+			);
+			const wtnLedger = new CanisterActor<icrcLedgerInterface>(
 				agent,
-				canisterId: CANISTER_ID_WTN_LEDGER
-			});
-			const waterNeuron: waterNeuronInterface = Actor.createActor(idlFactoryWaterNeuron, {
+				idlFactoryIcrc,
+				CANISTER_ID_WTN_LEDGER
+			);
+			const waterNeuron = new CanisterActor<waterNeuronInterface>(
 				agent,
-				canisterId: CANISTER_ID_WATER_NEURON
-			});
-			const boomerang: boomerangInterface = Actor.createActor(idlFactoryBoomerang, {
+				idlFactoryWaterNeuron,
+				CANISTER_ID_WATER_NEURON
+			);
+			const boomerang = new CanisterActor<boomerangInterface>(
 				agent,
-				canisterId: CANISTER_ID_BOOMERANG
-			});
-			const icpswapPool: icpswapPoolInterface = Actor.createActor(idlFactoryIcpswapPool, {
+				idlFactoryBoomerang,
+				CANISTER_ID_BOOMERANG
+			);
+			const icpswapPool = new CanisterActor<icpswapPoolInterface>(
 				agent,
-				canisterId: CANISTER_ID_ICPSWAP_POOL
-			});
+				idlFactoryIcpswapPool,
+				CANISTER_ID_ICPSWAP_POOL
+			);
+
+			if (authenticatedAgent) {
+				if (DEV && !isPlug) {
```

Comment:
why do you check if it's not plug?

Reply:
Because locally (DEV), the Plug transport does not like when I `fetchRootKey`